backend/internal/handlers/handlers.go

```diff
@@ -407,6 +415,173 @@ func (h *AuthHandler) ManualSignIn(c echo.Context) error {
 	return c.JSON(http.StatusOK, map[string]string{"token": token})
 }
 
+func (h *AuthHandler) ForgotPassword(c echo.Context) error {
+	c.Logger().Info("Received forgot password request")
+	req := &ForgotPasswordRequest{}
+	if err := c.Bind(req); err != nil {
+		return echo.NewHTTPError(http.StatusBadRequest, err.Error())
+	}
+	if err := c.Validate(req); err != nil {
+		return echo.NewHTTPError(http.StatusBadRequest, err.Error())
+	}
+
+	// Check if the user exists
+	u := &models.User{}
+	result := h.DB.Where("email = ?", req.Email).First(u)
+	if errors.Is(result.Error, gorm.ErrRecordNotFound) {
+		return echo.NewHTTPError(http.StatusNotFound, "User not found")
+	}
```

Comment:
I think from a security standpoint, its a good practise to acknowledge this operation and return something along the lines:
https://ux.stackexchange.com/a/87093

So bad actors cannot reverse engineer if a user exists in the DB